server/src/main/java/org/folio/metastorage/server/Storage.java

```diff
@@ -200,28 +197,31 @@ Future<Void> upsertGlobalRecord(SqlConnection conn, UUID sourceId,
       JsonObject globalRecord, JsonArray matchKeyConfigs) {
 
     final String localIdentifier = globalRecord.getString("localId");
+    if (localIdentifier == null) {
+      return Future.failedFuture("localId required");
+    }
     if (Boolean.TRUE.equals(globalRecord.getBoolean("delete"))) {
       return deleteBibRecord(conn, localIdentifier, sourceId);
     }
-    final JsonObject marcPayload = globalRecord.getJsonObject("marcPayload");
-    final JsonObject inventoryPayload = globalRecord.getJsonObject("inventoryPayload");
-    return upsertBibRecord(conn, localIdentifier, sourceId, marcPayload, inventoryPayload,
-        matchKeyConfigs);
+    final JsonObject payload = globalRecord.getJsonObject("payload");
+    if (payload == null) {
+      return Future.failedFuture("payload required");
+    }
+    return upsertBibRecord(conn, localIdentifier, sourceId, payload, matchKeyConfigs);
```

Comment:
@adamdickmeiss Can you rename Storage functions from xxxBibRecord/s to xxxGlobalRecord, e.g upsertGlobalRecord